.ci/update-github-milestone.sh

```diff
@@ -1,21 +1,58 @@
-#!/bin/bash
```

Comment:
this shell is used in the middle of development process, not actually a release.

it will be required to run it to update version in pom and milestone to correct version.

Example:
1) now we have 10.3.4-SNAPSHOT in pom.
2) we merge some PR that is for issue that is marked as "breaking compatibility"
3) build in master fail (if we finish https://github.com/checkstyle/contribution/issues/648)
4) maintianer review failure in CI and confirm that we need to change version to 10.4-SNAPSHOT
5) maintainer run action to update version in pom and in github milestone
6) ready to merge new PR ..... release will be considered to be done in some time in future, may in few weeks.

Reply:
so means that we need to have new action for this activity.
and may be in future publish_releasenotes_outside.yml  be merged in this new action, or may be allways stay separate.
lets keep it separate for now.

I do not know why but some gut filling tell me to not mix most critical actions in github and release notes. I do not want to fail release if some problem appeared in release notes generation/publishing.

Reply:
@romani The idea of making every action with specialised approach was my first thought but with review, I changed it to general.

But I am making a special action for all the above tasks or maybe we can create 2 shell scripts:

1. Closing and Creating new milestone.
    - Publishing Jar.

2. Creating issue in other Repo.

And running both of these scripts in one GitHub action?

Reply:
PR description updated: https://github.com/checkstyle/checkstyle/pull/12119#issue-1353314309

Reply:
I am ok with 2 scripts